components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -289,14 +289,20 @@ void ConversationHandler::BindUntrustedConversationUI(
 }
 
 void ConversationHandler::OnConversationMetadataUpdated() {
-  // Pass the updated data to archive content
   if (archive_content_) {
-    archive_content_->SetMetadata(
-        metadata_->associated_content->url.value_or(GURL()),
-        base::UTF8ToUTF16(metadata_->associated_content->title.value_or("")),
-        metadata_->associated_content->content_type ==
-            mojom::ContentType::VideoTranscript);
+    if (metadata_->associated_content) {
+      // Pass the updated data to archive content
+      archive_content_->SetMetadata(
+          metadata_->associated_content->url,
+          base::UTF8ToUTF16(metadata_->associated_content->title),
+          metadata_->associated_content->content_type ==
+              mojom::ContentType::VideoTranscript);
+    } else {
+      // TODO: Confirm with @petemill if this is the correct behavior
+      archive_content_->SetMetadata(GURL(), u"", false);
```

Comment:
I can't see how this is possible. But I guess we should delete the archive content